### PR TITLE
Enhance output and filtering options

### PR DESCRIPTION
**Pull request by sethcleveland**
- Added flags for filtering for disambig pages, xml namespaces, and
    extracted text length

  - Added flag for printing document revision

### CODE REVIEW

**Comment by sethcleveland** [comment]
These are some options we needed to identify "quality" wikipedia content. Attempted to keep the default behavior intact while providing cmd-line flags for the new options. Based on some experiments with the simple wikipedia dump, these options reduce extracted content by ~15% while not affecting performance. Let me know if you'd like anything done to improve this pull request.

File: WikiExtractor.py [hunk 1]
```diff
@@ -452,11 +493,13 @@ def extract(self, out):
         self.magicWords['currentday'] = time.strftime('%d')
         self.magicWords['currenthour'] = time.strftime('%H')
         self.magicWords['currenttime'] = time.strftime('%H:%M:%S')
-        text = self.clean()
+        text = [line.encode('utf-8') for line in compact(self.clean())]
         footer = "\n</doc>\n"
+        if sum(len(line) for line in text) < Extractor.min_text_length:
```
Comment (nathj07) [inline WikiExtractor.py]: do we have an option for something like `-1` or the flag not being set so we can still get everything. If folks don't want this restriction it should be possible to still get everything.

Comment (sethcleveland) [inline WikiExtractor.py]: The default value is 0. Which would filter empty docs. It can be changed to -1 to capture empty docs.

Comment (nathj07) [inline WikiExtractor.py]: I was thinking how does someone use this library and *not* filter on length. Maybe I missed something.

**Comment by sethcleveland** [comment]
@attardi do you have any feedback?

**Comment by attardi** [comment]
Merged.
Thank you.

**Pull request closed without merging**